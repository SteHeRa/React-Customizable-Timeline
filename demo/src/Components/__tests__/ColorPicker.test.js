import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ColorPicker from '../ColorPicker/ColorPicker';

describe('ColorPicker', () => {
  it('calls when changed', () => {
    const setColors = jest.fn();
    render(
      <ColorPicker
        colors={{ primaryColor: '#000000', secondaryColor: '#FFFFFF' }}
        setColors={setColors}
      />,
    );

    const primaryColor = screen.getAllByRole('textbox')[0];
    const secondaryColor = screen.getAllByRole('textbox')[1];
    const backgroundColor = screen.getAllByRole('textbox')[2];

    userEvent.type(primaryColor, 'test');
    userEvent.type(secondaryColor, 'test');
    //userEvent.type(backgroundColor, 'test'); GET ERROR "Cannot read property 'disabled' of undefined" when running this test???

    expect(setColors).toHaveBeenCalledTimes(8);
  });
});
