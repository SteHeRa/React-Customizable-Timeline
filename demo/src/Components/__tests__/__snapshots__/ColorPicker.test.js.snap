// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="color_section"
      >
        <div
          class="color_section_colors"
        >
          <h4>
             
            Primary
             
          </h4>
          <div
            style="display: flex;"
          >
            <input
              name="primaryColor"
              type="color"
              value="#000000"
            />
          </div>
          <input
            class="hex-code"
            name="primaryColor"
            placeholder="#000000"
            type="text"
            value="#000000"
          />
        </div>
        <div
          class="color_section_colors"
        >
          <h4>
             
            Secondary
             
          </h4>
          <div
            style="display: flex;"
          >
            <input
              name="secondaryColor"
              type="color"
              value="#FFFFFF"
            />
          </div>
          <input
            class="hex-code"
            name="secondaryColor"
            placeholder="#FFFFFF"
            type="text"
            value="#FFFFFF"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="color_section"
    >
      <div
        class="color_section_colors"
      >
        <h4>
           
          Primary
           
        </h4>
        <div
          style="display: flex;"
        >
          <input
            name="primaryColor"
            type="color"
            value="#000000"
          />
        </div>
        <input
          class="hex-code"
          name="primaryColor"
          placeholder="#000000"
          type="text"
          value="#000000"
        />
      </div>
      <div
        class="color_section_colors"
      >
        <h4>
           
          Secondary
           
        </h4>
        <div
          style="display: flex;"
        >
          <input
            name="secondaryColor"
            type="color"
            value="#FFFFFF"
          />
        </div>
        <input
          class="hex-code"
          name="secondaryColor"
          placeholder="#FFFFFF"
          type="text"
          value="#FFFFFF"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
